pubnub/lib/src/crypto/crypto.dart

```diff
@@ -1,132 +1,113 @@
+    legacyCryptoModule =
+        LegacyCryptoModule(defaultConfiguration: defaultConfiguration);
+  }
 
-  /// Whether a random IV should be used.
-  final bool useRandomInitializationVector;
+  factory CryptoModule.legacyCryptoModule(CipherKey cipherKey,
+      {defaultCryptoConfiguration = const CryptoConfiguration()}) {
+    return CryptoModule(
+        defaultCryptor: LegacyCryptor(cipherKey,
+            cryptoConfiguration: defaultCryptoConfiguration),
+        cryptors: <ICryptor>[AesCbcCryptor(cipherKey)]);
+  }
 
-  const CryptoConfiguration(
-      {this.encryptionMode = EncryptionMode.CBC,
-      this.encryptKey = true,
-      this.useRandomInitializationVector = true});
-}
+  factory CryptoModule.aesCbcCryptoModule(CipherKey cipherKey,
+      {defaultCryptoConfiguration = const CryptoConfiguration()}) {
+    return CryptoModule(
+        defaultCryptor: AesCbcCryptor(cipherKey),
+        cryptors: <ICryptor>[
+          LegacyCryptor(cipherKey,
+              cryptoConfiguration: defaultCryptoConfiguration),
+        ]);
+  }
 
-/// Default cryptography module used in PubNub SDK.
-class CryptoModule implements ICryptoModule {
-  final CryptoConfiguration defaultConfiguration;
-
-  /// Default configuration is:
-  /// * `encryptionMode` set to [EncryptionMode.CBC].
-  /// * `encryptKey` set to `true`.
-  /// * `useRandomInitializationVector` set to `true`.
-  CryptoModule({this.defaultConfiguration = const CryptoConfiguration()});
-
-  crypto.Key _getKey(CipherKey cipherKey, CryptoConfiguration configuration) {
-    if (configuration.encryptKey) {
-      return crypto.Key.fromUtf8(
-          sha256.convert(cipherKey.data).toString().substring(0, 32));
-    } else {
-      return crypto.Key(Uint8List.fromList(cipherKey.data));
-    }
+  @override
+  List<int> encrypt(List<int> data) {
+    _emptyContentValidation(data);
+    var encrypted = defaultCryptor.encrypt(data);
+    if (encrypted.metadata.isEmpty) return encrypted.data;
+
+    var header =
+        CryptorHeader.from(defaultCryptor.identifier, encrypted.metadata);
+    var headerData = List<int>.filled(header!.length, 0);
+    var pos = 0;
+    headerData.setAll(pos, header.data);
+    pos = header.length - encrypted.metadata.length;
+    headerData.setAll(pos, encrypted.metadata);
+    return [...headerData, ...encrypted.data];
   }
 
-  /// Decrypts [input] with [key] based on [configuration].
-  ///
-  /// If [configuration] is `null`, then [CryptoModule.defaultConfiguration] is used.
   @override
-  dynamic decrypt(CipherKey key, String input,
-      {CryptoConfiguration? configuration}) {
-    var config = configuration ?? defaultConfiguration;
+  List<int> decrypt(List<int> data) {
+    var header = CryptorHeader.tryParse(data);
+    var cryptor = _getCryptor(header);
+    var headerLength = header != null ? header.length : 0;
+    var metadata = headerLength > 0
+        ? data.sublist((headerLength - header!.metadataLength), headerLength)
```

Comment:
The same, are you sure that `header` is not null here?

Reply:
Yes, Because we already performed check : 
`var headerLength = header != null ? header.length : 0;`

we go with sublist only when we have `headerLength>0 that` means header is not null.